src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -418,17 +418,17 @@ private static bool IsDefinedSafe(MemberInfo memberInfo, Type attributeType, boo
             }
         }
 
-        private static Attribute[] GetCustomAttributesSafe(MemberInfo memberInfo, Type attributeType, bool inherit)
+        private static Attribute[]? GetCustomAttributesSafe(MemberInfo memberInfo, Type attributeType, bool inherit)
         {
             try
             {
                 return Attribute.GetCustomAttributes(memberInfo, attributeType, inherit);
             }
             catch
             {
-                // Getting the attributes has failed, return an empty array. One of the reasons
+                // Getting the attributes has failed, return null. One of the reasons
                 // can be that the member has attributes defined in an assembly that is missing.
-                return [];
+                return null;
```

Comment:
Caller casts this to `StateMachineAttribute[]`. Casting empty `Attribute[]` to `StateMachineAttribute[]` was guaranteed to fail.